### ISSUE TITLE
[Linking API] Unexpected result for extract-link-targets on presentations

### ISSUE THREAD

**Issue by juliushaertl** [CollaboraOnline/online#6230]
**Describe the bug**

The extract-link-targets endpoint for presentation files returns rather unexpected results. Taking this example file
[presentation.odp](https://github.com/CollaboraOnline/online/files/11323267/presentation.odp)

This is the output I get for the targets:

```
{
	"Targets": {
		"Handouts": true,
		"Slide 1": true,
		"Slide 1 (Notes)": true,
		"Slide 2": true,
		"Slide 2 (Notes)": true,
		"Slide 3": true,
		"Slide 3 (Notes)": true,
		"Slide 4": true,
		"Slide 4 (Notes)": true,
		"Slide 5": true,
		"Slide 5 (Notes)": true,
		"Slide 6": true,
		"Slide 6 (Notes)": true,
		"Slide 7": true,
		"Slide 7 (Notes)": true,
		"Slide 8": true,
		"Slide 8 (Notes)": true,
		"Slide 9": true,
		"Slide 9 (Notes)": true,
		"Slide 10": true,
		"Slide 10 (Notes)": true,
		"Slide 11": true,
		"Slide 11 (Notes)": true,
		"Slide 12": true,
		"Slide 12 (Notes)": true,
		"Title": true,
		"Title (Notes)": true,
		"One Bullet": true,
		"One Bullet (Notes)": true,
		"Two Bullets": true,
		"Two Bullets (Notes)": true,
		"Three Bullets": true,
		"Three Bullets (Notes)": true,
		"Four Bullets": true,
		"Four Bullets (Notes)": true,
		"Blue/White Chart": true,
		"Blue/White Chart (Notes)": true,
		"Table": true,
		"Table (Notes)": true,
		"White/Blue Chart": true,
		"White/Blue Chart (Notes)": true,
		"Right Bar": true,
		"Right Bar (Notes)": true,
		"Diagram": true,
		"Diagram (Notes)": true,
		"Left Bar": true,
		"Left Bar (Notes)": true
	}
}
```

However as discussing with @pedropintosilva I would rather expect to see different grouped types like Slides, Shapes, OLE Objects with their target identifiers.

For reference this is how the navigator tree looks like:

![Screenshot 2023-04-25 at 16 00 07](https://user-images.githubusercontent.com/3404133/234301217-2efe92d1-0789-4310-bab6-c577793eed81.png)

Tested with CODE 22.05.13.1

**Comment by eszkadev** [comment]
Impress doesn't support jumping to the items like "shape", it only supports jump to slides.
Fix improving the JSON so it will contain     `name: target` not `name: true`: https://gerrit.libreoffice.org/c/core/+/151884

**Comment by juliushaertl** [comment]
Thanks, i think just jumping to slides is perfectly fine and the most common use case.

**Comment by eszkadev** [comment]
patch merged to 23.05
